mailpoet/lib/Segments/SegmentsRepository.php

```diff
@@ -199,15 +209,15 @@ public function bulkDelete(array $ids, $type = SegmentEntity::TYPE_DEFAULT) {
         'ids' => $ids,
       ], ['ids' => Connection::PARAM_INT_ARRAY]);
 
-      return $entityManager->getConnection()->executeStatement("
-         DELETE s FROM $segmentTable s
-         WHERE s.`id` IN (:ids)
-         AND s.`type` = :type
-      ", [
-        'ids' => $ids,
-        'type' => $type,
-      ], ['ids' => Connection::PARAM_INT_ARRAY]);
+      $queryBuilder = $entityManager->createQueryBuilder();
+      $count = $queryBuilder->delete(SegmentEntity::class, 's')
```

Comment:
seems to work here, but would it make sense to typecast the return value of `->execute()`? 
![image](https://user-images.githubusercontent.com/6458412/169463327-fd6db626-013b-4bd6-91e7-6ef85e478aae.png)

Reply:
It should return an integer with the number of rows affected because it is a delete query. `execute` can return different types depending on the type of query but in this case, I think it is safe not to cast it: https://github.com/doctrine/orm/blob/8f7701279de84411020304f668cc8b49a5afbf5a/lib/Doctrine/ORM/AbstractQuery.php#L1355

Reply:
thanks for looking into it!